docs/migration-v3-v4.md

```diff
@@ -16,18 +16,27 @@ This guide consists of 4 parts:
    match the new inputs.
 3. [Configuration Migration Details](#configuration-migration-details): A
    detailed explanation of the changes you will need to make to your
-   configuration to update to version 4. 
-4. [Cluster Migration Steps](#cluster-migration-steps): Detailed instructions 
-   for migrating your EKS cluster to be managed by version 4. After you have 
-   updated your configuration, you will still need to take some additional 
-   manual steps to have Terraform upgrade and manage your existing EKS 
-   cluster with the new version 4 configuration. This step can be skipped if 
+   configuration to update to version 4.
+4. [Cluster Migration Steps](#cluster-migration-steps): Detailed instructions
+   for migrating your EKS cluster to be managed by version 4. After you have
+   updated your configuration, you will still need to take some additional
+   manual steps to have Terraform upgrade and manage your existing EKS
+   cluster with the new version 4 configuration. This step can be skipped if
    you can tolerate simply creating a new EKS cluster and deleting the old one.
 
 
 #### Usage notes
 
 > [!CAUTION]
+> In order for automatic conversions to take place, AWS requires that you
+> migrate in 2 steps: first to "API_AND_CONFIG_MAP" and then to "API".
+> In the migration steps documented here, we abandon the `aws-auth` ConfigMap
+> in place, with its existing contents, and add the new access control
+> entries. In order to remove any access granted by the `aws-auth` ConfigMap,
+> you must complete the migration to "API" mode. Even then, the `aws-auth`
+> ConfigMap will still exist, but it will be ignored. You can then delete it manually.
+
+> [!WARNING]
 > Hopefully, and likely, the following does not apply to you, but just in case:
```

Comment:
```suggestion
> [!WARNING]
> #### Usage of Kubernetes and Helm Providers May Not Work
> Hopefully, and likely, the following does not apply to you, but just in case:
```